TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementCheckers.cs

```diff
@@ -22,4 +23,51 @@ public static class SavepointStatementChecker
             }
         }
     }
+
+    internal static class ExecuteImmediateStatementChecker
+    {
+        public static void OnCodeElement(ExecuteImmediateStatement executeImmediateStatement, CodeElementsParser.ExecuteImmediateStatementContext context)
+        {
+            var expressionType = executeImmediateStatement.Expression?.ExpressionType;
+            switch (expressionType)
+            {
+                case SqlExpressionType.Variable:
+                    var variable = (SqlVariable) executeImmediateStatement.Expression;
+                    if (variable.Type != VariableType.HostVariable)
+                    {
+                        DiagnosticUtils.AddError(executeImmediateStatement,
+                            "The sql variable must be a host variable in EXECUTE IMMEDIATE statement",
+                            context);
+                    }
+                    else
+                    {
+                        if (((HostVariable) variable).IndicatorReference != null)
+                        {
+                            DiagnosticUtils.AddError(executeImmediateStatement,
+                                "An indicator variable must not be specified with a host variable in EXECUTE IMMEDIATE statement",
+                                context);
+                        }
+                    }
+
+                    break;
+
+                case SqlExpressionType.Constant:
+                    var constant = (SqlConstant)executeImmediateStatement.Expression;
+                    if (constant.Type != SqlConstantType.CharacterString)
+                    {
+                        DiagnosticUtils.AddError(executeImmediateStatement,
+                            "The constant must be Alphanumeric in EXECUTE IMMEDIATE statement",
+                            context);
+                    }
+                    break;
+                default:
+                DiagnosticUtils.AddError(executeImmediateStatement,
+                    "Only 2 types of expression are allowed in EXECUTE IMMEDIATE statement: variable or constant",
+                    context);
+                break;
```

Comment:
- Add a `case null: break` to avoid creating diagnostic for null expression (it will be done by ANTLR itself).
- As for the message, you can use 'Invalid expression found after EXECUTE IMMEDIATE, variable or string-expression was expected.'